cclib/io/qcjsonwriter.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2018, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""A writer for MolSSI quantum chemical JSON (QCJSON) files.
+"""
+
+import json
+
+from cclib.parser.utils import convertor
+
+from .cjsonwriter import CJSON as CJSONWriter, JSONIndentEncoder, NumpyAwareJSONEncoder
+
+
+class QCJSONWriter(CJSONWriter):
+    """A writer for QCSchema files."""
+
+    def __init__(self, ccdata, *args, **kwargs):
+        super(QCJSONWriter, self).__init__(ccdata, *args, **kwargs)
+
+    def as_dict(self):
+        qcschema_dict = dict()
+
+        qcschema_dict["schema_name"] = "qcschema_output"
+        qcschema_dict["schema_version"] = 1
+
+        qcschema_dict["molecule"] = dict()
+        qcschema_dict["molecule"]["schema_name"] = "qcschema_molecule"
+        qcschema_dict["molecule"]["schema_version"] = 2
+
+        # FIXME hack
+        qcschema_dict["driver"] = "energy"
+
+        metadata = self.ccdata.metadata
+
+        qcschema_dict["success"] = metadata["success"]
+
+        # FIXME
+        # if "keywords" in metadata:
+        #     qcschema_dict["keywords"] = set(metadata["keywords"])
+        # else:
+        qcschema_dict["keywords"] = dict()
+
+        qcschema_dict["model"] = dict()
+        # TODO methods should be an enum
+        if metadata["methods"]:
+            method = metadata["methods"][-1]
+        else:
+            # FIXME
+            method = "HF"
+        qcschema_dict["model"]["method"] = method
+
+        # FIXME
+        qcschema_dict["model"]["basis"] = ""
+
+        qcschema_dict["provenance"] = dict()
+        qcschema_dict["provenance"]["creator"] = metadata["package"]
+        qcschema_dict["provenance"]["version"] = metadata["package_version"]
+        # FIXME
+        qcschema_dict["provenance"]["routine"] = ""
```

Comment:
Would we want something to indicate that this was created through cclib rather than by the `metadata['package']` directly?

Reply:
In an ideal world, that's extra information that goes in the provenance in addition to the package. I can upload the provenance notes I took at the meeting, but I don't think the provenance substructure requires it right now.

I can try adding fields to the output that aren't part of the schema and see if it passes validation, I don't know if the validator ignores extra fields.

Also, this is a reminder that if we think something is missing from the schema, we (cclib) can request it to be added.